### DISCUSSION TITLE
add api reporting producer runninginfo to broker as consumer did

### DISCUSSION THREAD

**Discussion by francisoliverlee** [apache/rocketmq#6539] [Ideas]
report producer running info as consumer did,  take push consumer running info for example :
https://github.com/apache/rocketmq/blob/develop/client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPushConsumerImpl.java#L1408-L1451

**Comment by Abhijeetmishr** [comment]
@francisoliverlee where can I get more details on it to start with ?
I will appreciate any help/hint.

Regards

**Reply by francisoliverlee** [reply]
consumers implements a method consumerRunningInfo() which will be called when admin tool calls broker, broker calls every online-consumer's consumerRunningInfo().

producer can do the same.
![image](https://user-images.githubusercontent.com/5908412/230837808-76a16a3f-2bca-4249-a57f-5732551c4214.png)

**Reply by francisoliverlee** [reply]
> @francisoliverlee where can I get more details on it to start with ? I will appreciate any help/hint.
> 
> Regards

any details， email me: 1026203200@qq.com, 微信同qq号

**Comment by yp969803** [comment]
@francisoliverlee can you tell me the properties and methods producreRunningInfo class should contain
